### DISCUSSION TITLE
Creating conductivity with SimNIBS and Brainstorm

### DISCUSSION THREAD

**Discussion by michefabb** [simnibs/simnibs#397] [7.Creating head models] [answered]
Dear SimNIBS experts,

I am writing you to understand a point about tensors created with SimNIBS and Brainstorm. I understand this is a bit in the middle, so need to check with both yourself and Brainstorm team.
I ran `charm` on my subject using T1w and T2w and now would like to run `dwi2cond` using my subject DWI to create the conductivity tensors (as specified in [https://simnibs.github.io/simnibs/build/html/documentation/command_line/dwi2cond.html](url) ); then if I choose the "direct mapping" method for conductivity, would it be congruent with running the `EMA` method in Brainstorm (as specified in [https://neuroimage.usc.edu/brainstorm/Tutorials/FemTensors](url) ) using the same data (output from charm and DWI of the subject)? When I say "congruent" I mean if they produce similar results (even if not perfectly overlapping).
From the links I mentioned, it seems they are the only methods that refer to the same papers ([Rullmann et al., 2009](https://doi.org/10.1016/j.neuroimage.2008.09.009); [Tuch et al., 2001](https://doi.org/10.1073/pnas.171473898)).

Appreciated!

Mic

**Comment by bjornsig** [comment, marked as answer]
Yes, the basic principal is the same so they might be similar but we recommend testing it.

**Comment by michefabb** [comment]
@bjornsig running dwi2cond with eddy and reverse phase image
`dwi2cond --all --eddy --phasedir=y --readout=0.09027 sub-01_recon ${upper_level_dir}/sub-01_dwi.nii.gz ${upper_level_dir}/sub-01_dwi.bval ${upper_level_dir}/sub-01_dwi.bvec ${upper_level_dir}/b0_u.nii.gz`
At the end I get the following error:

> ==============================
==> --T1reg: Registration of preprocessed DWI data to T1 image ...
==============================
registering results to T1fs_conform
using nonlinear registration
Registering symmetric tensor...
/Users/michelangelo/Applications/SimNIBS-4.1/simnibs_env/lib/python3.9/site-packages/requests/__init__.py:102: RequestsDependencyWarning: urllib3 (1.26.18) or chardet (5.2.0)/charset_normalizer (2.0.4) doesn't match a supported version!
  warnings.warn("urllib3 ({}) or chardet ({})/charset_normalizer ({}) doesn't match a supported "
Traceback (most recent call last):
  File "/Users/michelangelo/Applications/SimNIBS-4.1/simnibs_env/lib/python3.9/site-packages/simnibs/cli/nii2msh.py", line 32, in <module>
    from ..utils.mesh_element_properties import ElementTags
ImportError: attempted relative import with no known parent package
exiting dwi2cond after error

Opening the dwi2cond_log file it states:

> nii2msh -ev /Users/michelangelo/Stella_Maris_Data/Stella_Maris_Patient_SS3T-CSD/SimNIBS/m2m_sub-01_recon/DTI_coregT1_tensor.nii.gz /Users/michelangelo/Stella_Maris_Data/Stella_Maris_Patient_SS3T-CSD/SimNIBS/m2m_sub-01_recon/sub-01_recon.msh /Users/michelangelo/Stella_Maris_Data/Stella_Maris_Patient_SS3T-CSD/SimNIBS/m2m_sub-01_recon/dMRI_prep/first_ev_for_check.msh
/Users/michelangelo/Applications/SimNIBS-4.1/simnibs_env/lib/python3.9/site-packages/requests/__init__.py:102: RequestsDependencyWarning: urllib3 (1.26.18) or chardet (5.2.0)/charset_normalizer (2.0.4) doesn't match a supported version!
  warnings.warn("urllib3 ({}) or chardet ({})/charset_normalizer ({}) doesn't match a supported "
Traceback (most recent call last):
  File "/Users/michelangelo/Applications/SimNIBS-4.1/simnibs_env/lib/python3.9/site-packages/simnibs/cli/nii2msh.py", line 32, in 
    from ..utils.mesh_element_properties import ElementTags
ImportError: attempted relative import with no known parent package
error code 1
exiting dwi2cond

Could it be a problem in the code? Or any other suggestion?

Thanks!

Mic

**Reply by michefabb** [reply]
Dear @bjornsig @jduemose ,

I think I found the issue: instead of `from ..utils.mesh_element_properties import ElementTags` it should be `from simnibs.utils.mesh_element_properties import ElementTags` which seems to be already fixed.
I am using SimNIBS 4.1.0a1.dev23+ge15afab5 , how can I update to the latest stable version 4.1.0 without creating issues?
Is this version able to use FS pial & white surfaces?

Thanks, regards,

Mic

**Reply by oulap** [reply]
Hi Mic,

If you installed simnibs from source using git, you can just pull the latest version and install it like you did previously. If you used the installer, just uninstall the previous version and download the latest installer from the "Releases"-page. The 4.x versions are backwards compatible, so you shouldn't enocunter any issues. In the latest version you can also use FS pial and wm surfaces using the --fs-dir flag in charm.

**Reply by michefabb** [reply]
Thanks @oulap ,

I uninstalled and installed with the .pkg, now I am trying to add spyder as I had with the 4.1.0a1 version, I just ran from Terminal:
simnibs_python -m pip install spyder==5.3.1

If I do it now it stacks during the process...any suggestion?

Thanks,

Mic

**Reply by oulap** [reply]
Hmm, I'm not sure I haven't used spyder that much. Is it trying to solve for the environment or retrieve the package? Where is it stuck exactly?

**Reply by michefabb** [reply]
Thanks @oulap , see below: during last line it continues it get stack... let me add I am using a Mac with M3

Collecting spyder
  Using cached spyder-5.5.5-py3-none-any.whl (14.9 MB)
Collecting chardet>=2.0.0
  Using cached chardet-5.2.0-py3-none-any.whl (199 kB)
Collecting nbconvert>=4.0
  Using cached nbconvert-7.16.4-py3-none-any.whl (257 kB)
Collecting rtree>=0.9.7
  Downloading Rtree-1.3.0-py3-none-macosx_10_9_x86_64.whl (475 kB)
     |████████████████████████████████| 475 kB 910 kB/s 
Collecting parso<0.9.0,>=0.7.0
  Downloading parso-0.8.4-py2.py3-none-any.whl (103 kB)
     |████████████████████████████████| 103 kB 35.4 MB/s 
Collecting qdarkstyle<3.3.0,>=3.2.0
  Using cached QDarkStyle-3.2.3-py2.py3-none-any.whl (871 kB)
Collecting pyls-spyder>=0.4.0
  Using cached pyls_spyder-0.4.0-py3-none-any.whl (8.1 kB)
Collecting pickleshare>=0.4
  Using cached pickleshare-0.7.5-py2.py3-none-any.whl (6.9 kB)
Collecting psutil>=5.3
  Using cached psutil-6.0.0-cp36-abi3-macosx_10_9_x86_64.whl (250 kB)
Collecting qtconsole<5.6.0,>=5.5.1
  Downloading qtconsole-5.5.2-py3-none-any.whl (123 kB)
     |████████████████████████████████| 123 kB 32.5 MB/s 
Collecting jellyfish>=0.7
  Using cached jellyfish-1.1.0-cp39-cp39-macosx_10_12_x86_64.whl (307 kB)
Collecting spyder-kernels<2.6.0,>=2.5.2
  Downloading spyder_kernels-2.5.2-py2.py3-none-any.whl (99 kB)
     |████████████████████████████████| 99 kB 4.3 MB/s 
Collecting watchdog>=0.10.3
  Downloading watchdog-4.0.1-cp39-cp39-macosx_10_9_x86_64.whl (92 kB)
     |████████████████████████████████| 92 kB 870 kB/s 
Requirement already satisfied: jsonschema>=3.2.0 in ./Applications/SimNIBS-4.1/simnibs_env/lib/python3.9/site-packages (from spyder) (4.17.3)
Collecting pexpect>=4.4.0
  Using cached pexpect-4.9.0-py2.py3-none-any.whl (63 kB)
Collecting python-lsp-black<3.0.0,>=2.0.0
  Using cached python_lsp_black-2.0.0-py3-none-any.whl (6.4 kB)
Collecting three-merge>=0.1.1
  Using cached three_merge-0.1.1-py2.py3-none-any.whl (6.4 kB)
Collecting ipython!=8.17.1,<9.0.0,>=8.13.0
  Using cached ipython-8.18.1-py3-none-any.whl (808 kB)
Collecting sphinx>=0.6.6
  Downloading sphinx-7.4.7-py3-none-any.whl (3.4 MB)
     |████████████████████████████████| 3.4 MB 582 kB/s 
Collecting cookiecutter>=1.6.0
  Using cached cookiecutter-2.6.0-py3-none-any.whl (39 kB)
Collecting pyzmq>=24.0.0
  Using cached pyzmq-26.1.0-cp39-cp39-macosx_10_15_universal2.whl (1.3 MB)
Collecting keyring>=17.0.0
  Using cached keyring-25.3.0-py3-none-any.whl (38 kB)
Collecting qstylizer>=0.2.2
  Downloading qstylizer-0.2.3-py2.py3-none-any.whl (15 kB)
Collecting pyqtwebengine<5.16,>=5.10
  Downloading PyQtWebEngine-5.15.7.tar.gz (32 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
    Preparing wheel metadata ... error
    **ERROR: Command errored out with exit status 1:
     command: /Users/michelangelo/Applications/SimNIBS-4.1/simnibs_env/bin/python /Users/michelangelo/Applications/SimNIBS-4.1/simnibs_env/lib/python3.9/site-packages/pip/_vendor/pep517/in_process/_in_process.py prepare_metadata_for_build_wheel /var/folders/0h/fw0yhv_948zc7mx84h27wbc40000gn/T/tmph0uz60e8
         cwd: /private/var/folders/0h/fw0yhv_948zc7mx84h27wbc40000gn/T/pip-install-md88irxf/pyqtwebengine_4c4ef7abe4434ad6ab899b26b1facfaa
    Complete output (7 lines):
    Querying qmake about your Qt installation...
    These bindings will be built: QtWebEngineCore, QtWebEngine, QtWebEngineWidgets.
    Generating the QtWebEngineCore bindings...
    _in_process.py: /private/var/folders/0h/fw0yhv_948zc7mx84h27wbc40000gn/T/pip-install-md88irxf/pyqtwebengine_4c4ef7abe4434ad6ab899b26b1facfaa/sip/QtWebEngineCore/QtWebEngineCoremod.sip: line 25: column 9: 'QtCore/QtCoremod.sip' could not be found
    /private/var/folders/0h/fw0yhv_948zc7mx84h27wbc40000gn/T/pip-install-md88irxf/pyqtwebengine_4c4ef7abe4434ad6ab899b26b1facfaa/sip/QtWebEngineCore/QtWebEngineCoremod.sip: line 26: column 9: 'QtGui/QtGuimod.sip' could not be found
    /private/var/folders/0h/fw0yhv_948zc7mx84h27wbc40000gn/T/pip-install-md88irxf/pyqtwebengine_4c4ef7abe4434ad6ab899b26b1facfaa/sip/QtWebEngineCore/QtWebEngineCoremod.sip: line 27: column 9: 'QtNetwork/QtNetworkmod.sip' could not be found
    /private/var/folders/0h/fw0yhv_948zc7mx84h27wbc40000gn/T/pip-install-md88irxf/pyqtwebengine_4c4ef7abe4434ad6ab899b26b1facfaa/sip/QtWebEngineCore/qwebengineclientcertificatestore.sip: line 24: column 6: 'PyQt_SSL' is not a known qualifier**
    ----------------------------------------
**WARNING: Discarding https://files.pythonhosted.org/packages/18/e8/19a00646866e950307f8cd73841575cdb92800ae14837d5821bcbb91392c/PyQtWebEngine-5.15.7.tar.gz#sha256=f121ac6e4a2f96ac289619bcfc37f64e68362f24a346553f5d6c42efa4228a4d (from https://pypi.org/simple/pyqtwebengine/) (requires-python:>=3.8). Command errored out with exit status 1: /Users/michelangelo/Applications/SimNIBS-4.1/simnibs_env/bin/python /Users/michelangelo/Applications/SimNIBS-4.1/simnibs_env/lib/python3.9/site-packages/pip/_vendor/pep517/in_process/_in_process.py prepare_metadata_for_build_wheel /var/folders/0h/fw0yhv_948zc7mx84h27wbc40000gn/T/tmph0uz60e8 Check the logs for full command output.**
  Using cached PyQtWebEngine-5.15.6-cp37-abi3-macosx_10_13_x86_64.whl (187 kB)
Collecting pylint-venv>=3.0.2
  Using cached pylint_venv-3.0.3-py3-none-any.whl (5.3 kB)
Collecting intervaltree>=3.0.2
  Using cached intervaltree-3.1.0-py2.py3-none-any.whl
Requirement already satisfied: setuptools>=49.6.0 in ./Applications/SimNIBS-4.1/simnibs_env/lib/python3.9/site-packages (from spyder) (68.2.2)
Collecting qtawesome<1.4.0,>=1.3.1
  Downloading QtAwesome-1.3.1-py3-none-any.whl (2.4 MB)
     |████████████████████████████████| 2.4 MB 36.8 MB/s 
Collecting cloudpickle>=0.5.0
  Using cached cloudpickle-3.0.0-py3-none-any.whl (20 kB)
Collecting numpydoc>=0.6.0
  Downloading numpydoc-1.7.0-py3-none-any.whl (62 kB)
     |████████████████████████████████| 62 kB 2.1 MB/s 
Collecting pylint<4,>=3.1
  Downloading pylint-3.2.6-py3-none-any.whl (519 kB)
     |████████████████████████████████| 519 kB 41.7 MB/s 
Collecting textdistance>=4.2.0
  Downloading textdistance-4.6.3-py3-none-any.whl (31 kB)
Collecting python-lsp-server[all]<1.12.0,>=1.11.0
  Downloading python_lsp_server-1.11.0-py3-none-any.whl (74 kB)
     |████████████████████████████████| 74 kB 2.9 MB/s 
Collecting qtpy>=2.1.0
  Using cached QtPy-2.4.1-py3-none-any.whl (93 kB)
Collecting jedi<0.20.0,>=0.17.2
  Using cached jedi-0.19.1-py2.py3-none-any.whl (1.6 MB)
Collecting pygments>=2.0
  Using cached pygments-2.18.0-py3-none-any.whl (1.2 MB)
Collecting diff-match-patch>=20181111
  Using cached diff_match_patch-20230430-py3-none-any.whl (42 kB)
Collecting atomicwrites>=1.2.0
  Using cached atomicwrites-1.4.1-py2.py3-none-any.whl
Collecting pyqt5<5.16,>=5.10
  Using cached PyQt5-5.15.11.tar.gz (3.2 MB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
    Preparing wheel metadata ... -

**Reply by oulap** [reply]
Seems to be a problem with pyqt (or pyqtwebengine). Can you try to update those either using pip or conda and see if that works? It might take a couple of tries to find the correct versions for the qt-packages and spyder.

**Reply by michefabb** [reply]
Hi @oulap , I fixed it following [https://github.com/simnibs/simnibs/discussions/379](url)

**Reply by dariasve20** [reply]
Hi SimNIBS team,

I was experiencing the same issue. I was trying to access the link above (https://github.com/simnibs/simnibs/discussions/379), but it is not available. Could you share how did you fix the issue?

==============================
==> --T1reg: Registration of preprocessed DWI data to T1 image ...
==============================
registering results to T1fs_conform
using nonlinear registration
Registering symmetric tensor...
Traceback (most recent call last):
  File "/Users/ariasd/Applications/SimNIBS-4.1/simnibs_env/lib/python3.9/site-packages/simnibs/cli/nii2msh.py", line 32, in <module>
    from ..utils.mesh_element_properties import ElementTags
ImportError: attempted relative import with no known parent package
exiting dwi2cond after error

**Reply by dariasve20** [reply]
My bad, I just got access to the link.

Thank you!
